<a href="https://colab.research.google.com/github/NEEL5252/Global-Data-Analysis/blob/main/Analyzing_of_the_global_terrorism_dataset.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **PROJECT NAME**: Global Terrorism Dataset

### **PROJECT TYPE:** EDA
### **CONTRIBUTOR:** Individual

# **PROJECT SUMMARY**

Using the Global Terrorism Database (GTD) from 1970 to 2017, this research investigates essential facets of global terrorism through extensive analysis. The findings include locations with various assault intensities, trends in annual attack frequency, success rates, and the top nations with both the highest and lowest fatalities. It dives into the number of injuries received in each nation, the number of fatalities and injuries per area, the weapons used, and the geographical predominance of assault-based assaults. This study strengthens policymaking, counterterrorism efforts, and academic research by providing greater insights into the varied processes of terrorism. Finally, it prepares stakeholders to improve global security measures, creating a proactive approach to effectively battling terrorism.


# GITHUB link:


https://github.com/NEEL5252

# Problem Statement:

1. Attacks overall by areas
2. Total assaults for the year, whether they went down or up.
3. The number of successful and unsuccessful assaults.
4. The top 10 nations in terms of both kill rates and kills per capita
5. The total number of victims injured in terrorist acts worldwide
6. The overall death toll by geographic region
7. The overall number of victims per location of injuries
8. The total number of weapons used 9. The location of all the assaults that took place.


# Buisseness Objective

# **LET'S START THE ANALYSIS**

## **PHASE - 1**


```
#  Importing libraries, importing dataset and read the dataset
```



In [2]:
# Importing necessary libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import plotly.express as px
import seaborn as sns
import plotly.graph_objects as go

In [3]:
# Mount the google drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [28]:
# Load the csv file / Read the csv file
globalTd = pd.read_csv("/content/drive/MyDrive/Global Terrarisom dataSet/Global Terrorism Data.csv", encoding='ISO-8859-1')

<ipython-input-28-6f2e09c616af>:2: DtypeWarning: Columns (4,6,31,33,61,62,63,76,79,90,92,94,96,114,115,121) have mixed types. Specify dtype option on import or set low_memory=False.
  globalTd = pd.read_csv("/content/drive/MyDrive/Global Terrarisom dataSet/Global Terrorism Data.csv", encoding='ISO-8859-1')


In [5]:
# Print the globalTd data
globalTd.head()

eventid  iyear  imonth  iday approxdate  extended resolution  country  \
0  197000000001   1970       7     2        NaN         0        NaN       58   
1  197000000002   1970       0     0        NaN         0        NaN      130   
2  197001000001   1970       1     0        NaN         0        NaN      160   
3  197001000002   1970       1     0        NaN         0        NaN       78   
4  197001000003   1970       1     0        NaN         0        NaN      101   

          country_txt  region  ... addnotes scite1 scite2  scite3  dbsource  \
0  Dominican Republic       2  ...      NaN    NaN    NaN     NaN      PGIS   
1              Mexico       1  ...      NaN    NaN    NaN     NaN      PGIS   
2         Philippines       5  ...      NaN    NaN    NaN     NaN      PGIS   
3              Greece       8  ...      NaN    NaN    NaN     NaN      PGIS   
4               Japan       4  ...      NaN    NaN    NaN     NaN      PGIS   

   INT_LOG  INT_IDEO INT_MISC INT_ANY  related  
0        0         0        0       0      NaN  
1        0         1        1       1      NaN  
2       -9        -9        1       1      NaN  
3       -9        -9        1       1      NaN  
4       -9        -9        1       1      NaN  

[5 rows x 135 columns]

In [6]:
# Shape of the globalTd dataset
globalTd.shape

(181691, 135)

In [7]:
# Count the null and missing values from the particular column of the dataset
nullcounts = globalTd.isnull().sum()
# Total null and missing counts from the dataset
totalNull = nullcounts.sum()
print(totalNull)

13853997


### Insights of the dataset:-


Together, there are 181691 rows and columns. 1,38,53,997 pieces of information are either missing or null in total. The year, kind, quantity of fatalities, nature of the weapon used, quantity of injuries, and city of the assault are among the details included in this data collection. Null values and missing values for several columns need to be removed. The wrangling and cleansing of the data will soon be finished.



## **PHASE - 2:-**


```
# Understanding the dataset
```



In [8]:
# Describing our dataset
globalTd.describe()

eventid          iyear         imonth           iday  \
count  1.816910e+05  181691.000000  181691.000000  181691.000000   
mean   2.002705e+11    2002.638997       6.467277      15.505644   
std    1.325957e+09      13.259430       3.388303       8.814045   
min    1.970000e+11    1970.000000       0.000000       0.000000   
25%    1.991021e+11    1991.000000       4.000000       8.000000   
50%    2.009022e+11    2009.000000       6.000000      15.000000   
75%    2.014081e+11    2014.000000       9.000000      23.000000   
max    2.017123e+11    2017.000000      12.000000      31.000000   

            extended        country         region       latitude  \
count  181691.000000  181691.000000  181691.000000  177135.000000   
mean        0.045346     131.968501       7.160938      23.498343   
std         0.208063     112.414535       2.933408      18.569242   
min         0.000000       4.000000       1.000000     -53.154613   
25%         0.000000      78.000000       5.000000      11.510046   
50%         0.000000      98.000000       6.000000      31.467463   
75%         0.000000     160.000000      10.000000      34.685087   
max         1.000000    1004.000000      12.000000      74.633553   

          longitude    specificity  ...     ransomamt   ransomamtus  \
count  1.771340e+05  181685.000000  ...  1.350000e+03  5.630000e+02   
mean  -4.586957e+02       1.451452  ...  3.172530e+06  5.784865e+05   
std    2.047790e+05       0.995430  ...  3.021157e+07  7.077924e+06   
min   -8.618590e+07       1.000000  ... -9.900000e+01 -9.900000e+01   
25%    4.545640e+00       1.000000  ...  0.000000e+00  0.000000e+00   
50%    4.324651e+01       1.000000  ...  1.500000e+04  0.000000e+00   
75%    6.871033e+01       1.000000  ...  4.000000e+05  0.000000e+00   
max    1.793667e+02       5.000000  ...  1.000000e+09  1.320000e+08   

         ransompaid  ransompaidus  hostkidoutcome     nreleased  \
count  7.740000e+02    552.000000    10991.000000  10400.000000   
mean   7.179437e+05    240.378623        4.629242    -29.018269   
std    1.014392e+07   2940.967293        2.035360     65.720119   
min   -9.900000e+01    -99.000000        1.000000    -99.000000   
25%   -9.900000e+01      0.000000        2.000000    -99.000000   
50%    0.000000e+00      0.000000        4.000000      0.000000   
75%    1.273412e+03      0.000000        7.000000      1.000000   
max    2.750000e+08  48000.000000        7.000000   2769.000000   

             INT_LOG       INT_IDEO       INT_MISC        INT_ANY  
count  181691.000000  181691.000000  181691.000000  181691.000000  
mean       -4.543731      -4.464398       0.090010      -3.945952  
std         4.543547       4.637152       0.568457       4.691325  
min        -9.000000      -9.000000      -9.000000      -9.000000  
25%        -9.000000      -9.000000       0.000000      -9.000000  
50%        -9.000000      -9.000000       0.000000       0.000000  
75%         0.000000       0.000000       0.000000       0.000000  
max         1.000000       1.000000       1.000000       1.000000  

[8 rows x 77 columns]

In [11]:
# Calculating total number of kills and injured people during terrorist attacks
globalTd[['nkill', 'nwound']].describe()

nkill         nwound
count  171378.000000  165380.000000
mean        2.403272       3.167668
std        11.545741      35.949392
min         0.000000       0.000000
25%         0.000000       0.000000
50%         0.000000       0.000000
75%         2.000000       2.000000
max      1570.000000    8191.000000

In [13]:
# Counting total success attacks
globalTd[globalTd['success'] == 1]['success'].sum()

161632

In [15]:
# Different types of weapons used in terrorist attacks
globalTd['weaptype1_txt'].unique()

array(['Unknown', 'Explosives', 'Incendiary', 'Firearms', 'Chemical',
       'Melee', 'Sabotage Equipment',
       'Vehicle (not to include vehicle-borne explosives, i.e., car or truck bombs)',
       'Fake Weapons', 'Radiological', 'Other', 'Biological'],
      dtype=object)

In [17]:
# Types of the attacks
globalTd['attacktype1_txt'].unique()

array(['Assassination', 'Hostage Taking (Kidnapping)',
       'Bombing/Explosion', 'Facility/Infrastructure Attack',
       'Armed Assault', 'Hijacking', 'Unknown', 'Unarmed Assault',
       'Hostage Taking (Barricade Incident)'], dtype=object)

### Description

From the above description:
1. Total 181691 terrorist attacks have been occured till 2017
2. In latitude and longitude we have around 4000 null values which shows that the we don't know the attacked area
3. Total 171378 people were lost their lives during terrorist attacks
4. Total 165380 people were injured during terrorist attacks
5. There are many columns which we don't need we will remove them in near future during the data wrangling phase
6. Out of 181691, 161632 attacks have been successfully executed and remains were unsuccessfull attacks
7. There are different types of weapons were used in terrorist attacks i.e. Explosives, firearms, chemical etc...
8. There are many different types of attacks were done in the past like Assassination, bombing, hijacking, kidnapping etc...

## **PHASE - 3**


```
Data Wrangling
```


1. Remove columns which will not used in our analyis, which contains more numbers of null and missing values

In [19]:
# Make list of columns which we want to delete
deleteColumns = []
for i in globalTd.columns:
  deleteColumns.append(i)

In [29]:
# columns list which are not gonna deleted (Based on these columns we will analysis the dataset)
notDeleteColumns = ['evenid', 'iyear', 'country', 'country_txt', 'region', 'region_txt',
                    'provstate', 'city', 'latitude', 'longitude', 'location',
                    'doubtterr', 'alternative', 'multiple','success',
                    'suicide', 'attacktype1', 'attacktype1_txt', 'targtype1',
                    'targtype1_txt', 'targsubtype1', 'targsubtype1_txt', 'corp1',
                    'weaptype1', 'weaptype1_txt', 'nkill', 'nwound']

# Converting it into array
notDeleteColumnsArray = np.array(notDeleteColumns)

In [30]:
# Deleting not necessary columns from the dataset
for i in deleteColumns:
  if (notDeleteColumnsArray[notDeleteColumnsArray == i]):
    pass
  else:
    globalTd.drop(i, inplace=True, axis=1)

<ipython-input-30-1aaa51b08a68>:3: DeprecationWarning: The truth value of an empty array is ambiguous. Returning False, but in future this will result in an error. Use `array.size > 0` to check that an array is not empty.
  if (notDeleteColumnsArray[notDeleteColumnsArray == i]):
<ipython-input-30-1aaa51b08a68>:3: DeprecationWarning: The truth value of an empty array is ambiguous. Returning False, but in future this will result in an error. Use `array.size > 0` to check that an array is not empty.
  if (notDeleteColumnsArray[notDeleteColumnsArray == i]):
<ipython-input-30-1aaa51b08a68>:3: DeprecationWarning: The truth value of an empty array is ambiguous. Returning False, but in future this will result in an error. Use `array.size > 0` to check that an array is not empty.
  if (notDeleteColumnsArray[notDeleteColumnsArray == i]):
<ipython-input-30-1aaa51b08a68>:3: DeprecationWarning: The truth value of an empty array is ambiguous. Returning False, but in future this will result in an err

In [31]:
globalTd.head()

iyear  country         country_txt  region                   region_txt  \
0   1970       58  Dominican Republic       2  Central America & Caribbean   
1   1970      130              Mexico       1                North America   
2   1970      160         Philippines       5               Southeast Asia   
3   1970       78              Greece       8               Western Europe   
4   1970      101               Japan       4                    East Asia   

  provstate           city   latitude   longitude location  ...  \
0       NaN  Santo Domingo  18.456792  -69.951164      NaN  ...   
1   Federal    Mexico city  19.371887  -99.086624      NaN  ...   
2    Tarlac        Unknown  15.478598  120.599741      NaN  ...   
3    Attica         Athens  37.997490   23.762728      NaN  ...   
4   Fukouka        Fukouka  33.580412  130.396361      NaN  ...   

                  attacktype1_txt  targtype1                targtype1_txt  \
0                   Assassination         14  Private Citizens & Property   
1     Hostage Taking (Kidnapping)          7      Government (Diplomatic)   
2                   Assassination         10          Journalists & Media   
3               Bombing/Explosion          7      Government (Diplomatic)   
4  Facility/Infrastructure Attack          7      Government (Diplomatic)   

   targsubtype1                                   targsubtype1_txt  \
0          68.0                                     Named Civilian   
1          45.0  Diplomatic Personnel (outside of embassy, cons...   
2          54.0                    Radio Journalist/Staff/Facility   
3          46.0                                  Embassy/Consulate   
4          46.0                                  Embassy/Consulate   

                         corp1 weaptype1  weaptype1_txt nkill  nwound  
0                          NaN        13        Unknown   1.0     0.0  
1  Belgian Ambassador Daughter        13        Unknown   0.0     0.0  
2             Voice of America        13        Unknown   1.0     0.0  
3                          NaN         6     Explosives   NaN     NaN  
4                          NaN         8     Incendiary   NaN     NaN  

[5 rows x 26 columns]

All the columns that won't be needed in subsequent analysis have been eliminated.
Some columns had a significant amount of null entries, while others had the same data as other columns that had been erased.


## **PHASE - 4**


```
# DATA VISUALIZATION PART
```

